src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -113,10 +117,10 @@ void CoreWorkerDirectTaskSubmitter::HandleWorkerLeaseGranted(
 void CoreWorkerDirectTaskSubmitter::OnWorkerIdle(const WorkerAddress &addr,
                                                  bool was_error) {
   absl::MutexLock lock(&mu_);
-  if (queued_tasks_.empty() || was_error) {
-    auto lease_client = std::move(worker_to_lease_client_[addr]);
+  auto entry = worker_to_lease_client_[addr];
+  if (current_time_ms() > entry.second || queued_tasks_.empty() || was_error) {
```

Comment:
Consider moving the time check to the end, since it is the slowest expression to evaluate.